vtex/vtexis-compatibility-layer
# Add function to convert search-document info into Product

#### What problem is this solving?



#### How should this be manually tested?

You can compare the [intelligent-search-api response](https://thalyta--carrefourbr.myvtex.com/_v/api/intelligent-search/product?id=326782298&type=sku.id&salesChannel=1) (which uses this new function) with the [graphql response for product](https://thalyta--carrefourbr.myvtex.com/admin/graphql-ide) (that currently uses catalog-api results)

Graphql query you can use to check the fields:
```
query{
  product(identifier:{ field: sku, value: 326782298 }, salesChannel: 1) {
    brand
    brandId
    cacheId
    categories
    categoryId
    clusterHighlights{
      id
      name
    }
    description
    items {
      attachments{
        domainValues{
          FieldName
          MaxCaracters
          DomainValues
        }
        id
        required
      }
      complementName
      ean
      images{
        imageId
        imageTag
        imageUrl
        imageText
        imageLabel
      }
      itemId
      kitItems{
        itemId
        amount
      }
      measurementUnit
      name
      nameComplete
      referenceId{
        Key
        Value
      }
      sellers{
        addToCartLink
        commertialOffer{
          AvailableQuantity
          CacheVersionUsedToCallCheckout
          Installments{
            InterestRate
            Name
            NumberOfInstallments
            PaymentSystemGroupName
            PaymentSystemName
            TotalValuePlusInterestRate
            Value
          }
          ListPrice
          Price
          PriceValidUntil
          PriceWithoutDiscount
          RewardValue
          Tax
          discountHighlights{
            name
            additionalInfo{
              key
              value
            }
          }
          spotPrice
          taxPercentage
          teasers{
            name
            generalValues{
              key
              value
            }
            effects{
              parameters{
                name
                value
              }
            }
            conditions{
              parameters{
                name
                value
              }
            }
          }
        }
        sellerDefault
        sellerId
        sellerName
      }
      unitMultiplier
      variations{
        originalName
        name
        values
      }
      videos{
        videoUrl
      }
    }
  	jsonSpecifications
    link
    linkText
    priceRange{
      listPrice {
        lowPrice
        highPrice
      }
      sellingPrice{
        lowPrice
        highPrice
      }
    }
    productClusters{
      id
      name
    }
    productId
    productName
    productReference
    properties{
      name
      originalName
      values
    }
    releaseDate
    selectedProperties{
      key
      value
    }
    skuSpecifications {
      field {
        originalName
        name
      }
    }
    specificationGroups{
      name
      originalName
      specifications{
        name
        originalName
        values
      }
    }
  }
}

```

![long cat](https://media.giphy.com/media/v1.Y2lkPTc5MGI3NjExbmVuY3k0djFibTNhbG80N204amtvMTh5MnJuMDBkdzhxZ2lyZzBoYyZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/aYZIeUDU3qvPa/giphy.gif)

#### Screenshots or example usage

#### Type of changes


✔️ | Type of Change
---|---
_ | Bug fix 
✔️ | New feature 
_ | Breaking change 
_ | Technical improvements
